lms/djangoapps/discussion_api/api.py

```diff
@@ -378,11 +402,19 @@ def update_thread(request, thread_id, update_data):
         detail.
     """
     cc_thread, context = _get_thread_and_context(request, thread_id)
-    is_author = str(request.user.id) == cc_thread["user_id"]
-    if not (context["is_requester_privileged"] or is_author):
-        raise PermissionDenied()
+    editable_fields = _get_thread_editable_fields(cc_thread, context)
+    non_allowed_fields = set(update_data.keys()) - editable_fields
```

Comment:
set is necessary to use the subtraction operator. This could alternately be done with `[field for field in update_data.keys() if field not in editable_fields]` if you think that would be clearer.